packages/hoppscotch-common/src/components/collections/ImportExport.vue

```diff
@@ -1,361 +1,475 @@
       return
     }
 
-    stepResults.value.push(content)
-    hasFile.value = !!content?.length
-  }
+    if (E.isRight(collectionJSON))
+      collectionsGistExporter(collectionJSON.right, accessToken)
 
-  reader.readAsText(inputFileToImport.files[0])
+    isHoppGistCollectionExporterInProgress.value = false
+  },
 }
 
-const fileImported = () => {
-  toast.success(t("state.file_imported").toString())
-  hideModal()
-}
-const failedImport = () => {
-  toast.error(t("import.failed").toString())
-}
-const hideModal = () => {
-  resetImport()
-  emit("hide-modal")
-}
+const importerModules = computed(() => {
+  const enabledImporters = [
+    HoppRESTImporter,
+    HoppMyCollectionImporter,
+    HoppOpenAPIImporter,
+    HoppPostmanImporter,
+    HoppInsomniaImporter,
+    HoppGistImporter,
+  ]
+
+  const isTeams = props.collectionsType.type === "team-collections"
+
+  return enabledImporters.filter((importer) => {
+    return isTeams
+      ? importer.metadata.applicableTo.includes("team-workspace")
+      : importer.metadata.applicableTo.includes("personal-workspace")
+  })
+})
+
+const exporterModules = computed(() => {
+  const enabledExporters = [
+    HoppMyCollectionsExporter,
+    HoppTeamCollectionsExporter,
+  ]
+
+  if (platform.platformFeatureFlags.exportAsGIST) {
+    enabledExporters.push(HoppGistCollectionsExporter)
+  }
+
+  return enabledExporters.filter((exporter) => {
+    return exporter.metadata.applicableTo.includes(
+      props.collectionsType.type == "my-collections"
+        ? "personal-workspace"
+        : "team-workspace"
+    )
+  })
+})
+
+const hasTeamWriteAccess = computed(() => {
+  const { collectionsType } = props
 
-const resetImport = () => {
-  importerType.value = null
-  hasFile.value = false
-  hasGist.value = false
-  stepResults.value = []
-  inputChooseFileToImportFrom.value = ""
-  inputChooseGistToImportFrom.value = ""
-  mySelectedCollectionID.value = undefined
+  const isTeamCollection = collectionsType.type === "team-collections"
+
+  return isTeamCollection
+    ? collectionsType.selectedTeam?.myRole == "EDITOR" ||
+        collectionsType.selectedTeam?.myRole == "OWNER"
+    : false
+})
+
+const selectedTeamID = computed(() => {
+  const { collectionsType } = props
+
+  return collectionsType.type == "team-collections"
+    ? collectionsType.selectedTeam?.id
+    : undefined
+})
+
+const myCollections = useReadonlyStream(restCollections$, [])
+
+const getCollectionJSON = async () => {
+  let collections
+
+  if (
+    props.collectionsType.type == "team-collections" &&
+    props.collectionsType.selectedTeam?.id
+  ) {
+    const res = await getTeamCollectionJSON(
+      props.collectionsType.selectedTeam?.id
+    )
+
+    return E.isRight(res)
+      ? E.right(res.right.exportCollectionsToJSON)
+      : E.left(res.left)
+  } else {
+    collections = JSON.stringify(myCollections.value, null, 2)
+
+    return E.right(collections)
+  }
```

Comment:
Can there be a case where `selectedTeam.id` doesn't yield a value where it goes to the `else` block despite being in the team workspace? In that case, don't we require a fallback instead of the control going to the `my-collections` block?
Also, the `else` block is redundant since we're returning early. We can inline the JSON stringified collections object instead of assigning it to a variable since there aren't multiple references.